x/clp/keeper/liquidityprovider.go

```diff
@@ -16,6 +16,10 @@ func (k Keeper) SetLiquidityProvider(ctx sdk.Context, lp *types.LiquidityProvide
 	if !lp.Validate() {
 		return
 	}
+	// if lp last updated block is 0, set it to current block height
+	if lp.LastUpdatedBlock == 0 {
+		lp.LastUpdatedBlock = ctx.BlockHeight()
```

Comment:
Wouldn't you want to always update LastUpdatedBlock?

Reply:
Also how would you want this to be affected on distribution?

Reply:
@timlind good catch, I added the `LastUpdatedBlock` update to the `AddLiquidity` keeper function